Applications/Utils/ModelPreparation/createNeumannBc.cpp

```diff
@@ -0,0 +1,192 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include <tclap/CmdLine.h>
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+#include "GeoLib/GEOObjects.h"
+#include "GeoLib/IO/writeGeometryToFile.h"
+
+#include "MeshGeoToolsLib/MeshNodeSearcher.h"
+#include "MeshLib/IO/readMeshFromFile.h"
+#include "MeshLib/IO/writeMeshToFile.h"
+#include "MeshLib/convertMeshToGeo.h"
+#include "MeshLib/Mesh.h"
+#include "MeshLib/MeshSubsets.h"
+#include "MeshLib/MeshSurfaceExtraction.h"
+
+#include "NumLib/DOF/LocalToGlobalIndexMap.h"
+
+#include "ProcessLib/ProcessVariable.h"
+int main(int argc, char* argv[])
+{
+    ApplicationsLib::LogogSetup logo_setup;
+
+    TCLAP::CmdLine cmd(
+        "Integrates the given element property and outputs an OGS-5 direct "
+        "Neumann boundary condition",
+        ' ',
+        "0.1");
+
+    TCLAP::ValueArg<std::string> in_mesh("i",
+                                         "in-mesh",
+                                         "the mesh that has an element "
+                                         "property for the Neumann "
+                                         "boundary condition",
+                                         true,
+                                         "",
+                                         "filename for mesh input");
+    cmd.add(in_mesh);
+    TCLAP::ValueArg<std::string> property_arg(
+        "p",
+        "property-name",
+        "name of property for the Neumann boundary condition",
+        true,
+        "",
+        "string (property name)");
+    cmd.add(property_arg);
+
+    TCLAP::ValueArg<std::string> result_file(
+        "o",
+        "result-out",
+        "the file name the result will be written to ",
+        true,
+        "",
+        "output file name");
+    cmd.add(result_file);
+    cmd.parse( argc, argv );
+
+    // read surface mesh
+    std::unique_ptr<MeshLib::Mesh> surface_mesh(
+        MeshLib::IO::readMeshFromFile(in_mesh.getValue()));
+
+    std::string const prop_name("OriginalSubsurfaceNodeIDs");
+    boost::optional<MeshLib::PropertyVector<std::size_t> const&> node_id_pv(
+        surface_mesh->getProperties().getPropertyVector<std::size_t>(prop_name));
```

Comment:
It looks like the given mesh must have been created with a special tool. Should the name of that tool be mentioned somewhere?